src/stacks-hierarchy/Root_PurchaseHarborSearchScreen/HarborResults.tsx

```diff
@@ -0,0 +1,165 @@
+import {screenReaderPause} from '@atb/components/text';
+import {ThemeText} from '@atb/components/text';
+import {StyleSheet} from '@atb/theme';
+import {useTranslation} from '@atb/translations';
+import {insets} from '@atb/utils/insets';
+import React, {useEffect, useState} from 'react';
+import {TouchableOpacity, View} from 'react-native';
+import {ThemeIcon} from '@atb/components/theme-icon';
+import {Boat} from '@atb/assets/svg/mono-icons/transportation';
+import {StopPlace, StopPlaces} from '@atb/api/types/stopPlaces';
+import HarborSearchTexts from '@atb/translations/screens/subscreens/HarborSearch';
+import sortBy from 'lodash.sortby';
+import {GeoLocation} from '@atb/favorites';
+import haversine from 'haversine-distance';
+import {MessageBox} from '@atb/components/message-box';
+import {useGeolocationState} from '@atb/GeolocationContext';
+import {TFunc} from '@leile/lobo-t';
+
+type Props = {
+  harbors: StopPlaces;
+  onSelect: (l: StopPlace) => void;
+  searchText?: string;
+  fromHarborName?: string;
+};
+
+export const HarborResults: React.FC<Props> = ({
+  harbors,
+  onSelect,
+  searchText = '',
+  fromHarborName,
+}) => {
+  const styles = useThemeStyles();
+  const {t} = useTranslation();
+
+  const [harborResults, setHarborResults] = useState<StopPlaces | []>([]);
+  const {location} = useGeolocationState();
+  const currentLocation = location || undefined;
+
+  useEffect(() => {
+    harbors && setHarborResults(sortHarbors(harbors, currentLocation));
+  }, [harbors, currentLocation]);
+
+  useEffect(() => {
+    if (searchText.length > 1 && harbors) {
+      setHarborResults(
+        sortHarbors(harbors, currentLocation).filter((harbor) => {
+          return harbor.name.toLowerCase().includes(searchText.toLowerCase());
+        }),
+      );
+    } else {
+      harbors && setHarborResults(sortHarbors(harbors, currentLocation));
+    }
+  }, [searchText]);
```

Comment:
I think sorting by `currentLocation` can just be done on render, which should simplify this a little.

https://react.dev/learn/you-might-not-need-an-effect#how-to-tell-if-a-calculation-is-expensive